kuma/wiki/utils.py

```diff
@@ -59,3 +67,42 @@ def tidy_content(content):
         return tidied.decode('utf-8'), errors
     else:
         return content
+
+
+def analytics_user_counts(*revs):
+    """Given some document revision numbers, returns a dict matching those
+    with the number of users Google Analytics thinks has visited each revision.
+
+    """
+
+    scopes = ['https://www.googleapis.com/auth/analytics.readonly']
+    credentials = ServiceAccountCredentials.from_json_keyfile_dict(
+        json.loads(config.GOOGLE_ANALYTICS_CREDENTIALS), scopes=scopes)
```

Comment:
There are at least three things that can go wrong with this one line:

1. The user hasn't customized ``GOOGLE_ANALYTICS_CREDENTIALS``, and it deserializes as an empty dictionary
2. The user has customized it, but with invalid JSON, like ``{bad json]``.
3. The user has valid looking credentials that Google doesn't like.

The first case has a traceback that looks like:
```
>>> analytics_user_counts(rev.id)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/home/vagrant/src/kuma/wiki/utils.py", line 80, in analytics_user_counts
    json.loads(config.GOOGLE_ANALYTICS_CREDENTIALS), scopes=scopes)
  File "/home/vagrant/env/local/lib/python2.7/site-packages/oauth2client/service_account.py", line 254, in from_json_keyfile_dict
    revoke_uri=revoke_uri)
  File "/home/vagrant/env/local/lib/python2.7/site-packages/oauth2client/service_account.py", line 173, in _from_parsed_json_keyfile
    'Expected', client.SERVICE_ACCOUNT)
ValueError: ('Unexpected credentials type', None, 'Expected', 'service_account')
```

The second case has a traceback like this:
```
>>> analytics_user_counts(100)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/home/vagrant/src/kuma/wiki/utils.py", line 80, in analytics_user_counts
    json.loads(config.GOOGLE_ANALYTICS_CREDENTIALS), scopes=scopes)
  File "/usr/lib/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python2.7/json/decoder.py", line 382, in raw_decode
    obj, end = self.scan_once(s, idx)
ValueError: Expecting property name enclosed in double quotes: line 1 column 2 (char 1)
```

For the third case, I created a new private key pair, made sure it worked, then deleted it in the Google console.  Here's the error from that:

```
>>> analytics_user_counts('1096253')
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/home/vagrant/src/kuma/wiki/utils.py", line 105, in analytics_user_counts
    response = request.execute()
  File "/home/vagrant/env/local/lib/python2.7/site-packages/oauth2client/util.py", line 137, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "/home/vagrant/env/local/lib/python2.7/site-packages/googleapiclient/http.py", line 827, in execute
    method=str(self.method), body=self.body, headers=self.headers)
  File "/home/vagrant/env/local/lib/python2.7/site-packages/googleapiclient/http.py", line 154, in _retry_request
    resp, content = http.request(uri, method, *args, **kwargs)
  File "/home/vagrant/env/local/lib/python2.7/site-packages/oauth2client/transport.py", line 153, in new_request
    credentials._refresh(orig_request_method)
  File "/home/vagrant/env/local/lib/python2.7/site-packages/oauth2client/client.py", line 765, in _refresh
    self._do_refresh_request(http_request)
  File "/home/vagrant/env/local/lib/python2.7/site-packages/oauth2client/client.py", line 834, in _do_refresh_request
    raise HttpAccessTokenRefreshError(error_msg, status=resp.status)
HttpAccessTokenRefreshError: invalid_grant: Invalid JWT Signature.
```

I think the third error is OK at it is.  I'd like to distinguish between the first two.  I think a ``ValueError`` is right for badly written JSON, but I'd like a different, app-specific exception if the user hasn't configured ``GOOGLE_ANALYTICS_CREDENTIALS``.